### PR TITLE
assorted: use GetArgumentFromQueryString and other minor fixes

### PR DESCRIPTION
**Pull request by mynameisbogdan**
- use `ParseUtil.GetArgumentFromQueryString()` instead of `Split` (there a few more, but I can't test or find out what's the name of param)
- use [NameValueCollection.Set()](https://learn.microsoft.com/en-us/dotnet/api/system.collections.specialized.namevaluecollection.set?view=net-6.0#remarks)
- use `WebUtility.HtmlDecode()` for titles taken from download links
- ignore the hardcoded timezone in HDTS
- ST: use the date added before falling back to elapsed time, and change encoding to UTF-8 to match both header and meta `content-type` charset.

### CODE REVIEW

File: src/Jackett.Common/Indexers/SceneTime.cs [hunk 1]
```diff
@@ -172,43 +171,43 @@ private List<ReleaseInfo> ParseResponse(TorznabQuery query, string htmlResponse)
...
-                        MinimumSeedTime = 259200 // 72 hours
-                    };
-
-                    var catId = "82"; // default
-                    var qCatLink = row.Children[categoryIndex].QuerySelector("a");
-                    if (qCatLink != null)
-                    {
-                        catId = new Regex(@"\?cat=(\d*)").Match(qCatLink.GetAttribute("href")).Groups[1].ToString().Trim();
-                    }
-                    release.Category = MapTrackerCatToNewznab(catId);
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Indexers/SceneTime.cs:188]: This was added in https://github.com/Jackett/Jackett/commit/bef73a5f50dfef9739571bb1b1a2f07675d0caa5 to handle missing categories. It appears to be still be an issue. If you search `blue` you get the following error:

```text
2023-01-29 05:40:17 | Error | Error on 20230129054017 for SceneTime.txtSystem.NullReferenceException: Object reference not set to an instance of an object.    at Jackett.Common.Indexers.SceneTime.ParseResponse(TorznabQuery query, String htmlResponse) in D:\a\1\s\src\Jackett.Common\Indexers\SceneTime.cs:line 183
2023-01-29 05:40:17 | Error | Jackett.Common.IndexerException: Exception (scenetime): Object reference not set to an instance of an object.
```

**Review by ilike2burnthing — CHANGES_REQUESTED** [review]
HDSpace and ImmortalSeed appear to be fine, just the one issue with SceneTime.

@garfield69 can you test XSpeeds?

The changes to HDTorrents seem fairly simple, don't think there's any need to ask ngosang to test it.

**Review by ilike2burnthing — APPROVED** [review]

**Comment by garfield69** [comment]
>use ParseUtil.GetArgumentFromQueryString() instead of Split (there a few more, but I can't test or find out what's the name of param)

if you can identify which other indexers you are referring to, I may be able to provide temp access while you code/test (if I have an account)

**Comment by garfield69** [comment]
v0.20.2767

**Pull request merged**